### PR TITLE
make public workspaces optional

### PR DESCRIPTION
**Pull request by j8seangel**
Replace the window.alert in workspace save for the following modal to add the public option:

image

and disable it when private datasets are in use:
image

### CODE REVIEW

File: applications/fishing-map/src/features/sidebar/SidebarHeader.tsx [hunk 1]
```diff
@@ -21,94 +17,70 @@ import { AsyncReducerStatus } from 'utils/async-slice'
...
   const workspaceCustomStatus = useSelector(selectWorkspaceCustomStatus)
   const { showClipboardNotification, copyToClipboard } = useClipboardNotification()
   const workspace = useSelector(selectWorkspace)
 
   const isOwnerWorkspace = workspace?.ownerId === userData?.id
-  const isDefaultWorkspace = workspace?.id === DEFAULT_WORKSPACE_ID
 
-  const updateWorkspaceAction = (workspaceId: string) => {
-    return dispatch(updatedCurrentWorkspaceThunk(workspaceId))
+  const onCloseCreateWorkspace = () => {
```
Comment (nerik) [inline applications/fishing-map/src/features/sidebar/SidebarHeader.tsx:44]: Shouldn't we use `useCallback` ?

File: applications/fishing-map/src/features/workspace/shared/NewWorkspaceModal.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      const start = formatI18nDate(timerange.start as string, {
+        format: dateFormat,
+      })
+        .replace(',', '')
+        .replace('.', '')
+      const end = formatI18nDate(timerange.end as string, {
+        format: dateFormat,
+      })
+        .replace(',', '')
+        .replace('.', '')
```
Comment (nerik) [inline applications/fishing-map/src/features/workspace/shared/NewWorkspaceModal.tsx:50]:
```suggestion
      const formatTimerangeBoundary = (boundary?: string) => {
        return formatI18nDate(boundary as string, {
        format: dateFormat,
      })
        .replace(/[\.\,]/g, '')
      }
      const start = formatTimerangeBoundary(timerange.start)
      const end = formatTimerangeBoundary(timerange.end)
```

File: applications/fishing-map/src/features/workspace/shared/NewWorkspaceModal.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      onClose={onClose}
+    >
+      <InputText
+        inputSize="small"
+        value={name}
+        label={t('common.name', 'Name')}
+        className={styles.input}
+        onChange={(e) => setName(e.target.value)}
+      />
+      <Radio
```
Comment (nerik) [inline applications/fishing-map/src/features/workspace/shared/NewWorkspaceModal.tsx:98]: This must be a `Checkbox`, not a `Radio`

**Review by nerik — CHANGES_REQUESTED** [review]
Looks good! Please see my comments.

I have doubts about the flow: when a user saves a workspace once, the following saves won't trigger the modal again. Which means there is no way to change the public visibility setting again, nor changing the title (which means you can completely change dates for instance and still have the old dates in the title). Is that intended? How do we allow the user to create a new workspace from an existing one, rather than updating it?

File: applications/fishing-map/src/features/workspace/shared/NewWorkspaceModal.tsx (hunk 2 above)
Comment (j8seangel) [reply applications/fishing-map/src/features/workspace/shared/NewWorkspaceModal.tsx]: Thanks, added in https://github.com/GlobalFishingWatch/frontend/commit/264d9caefaa5e320a92028293a3f2e1b6d1adb75

File: applications/fishing-map/src/features/workspace/shared/NewWorkspaceModal.tsx (hunk 3 above)
Comment (j8seangel) [reply applications/fishing-map/src/features/workspace/shared/NewWorkspaceModal.tsx]: Replaced by a Switch, we don't have a Checkbox component yet

**Pull request merged**